src/lib/src/component/navigation/navigation.component.spec.ts

```diff
@@ -515,4 +521,209 @@ describe('NavigationComponent', () => {
     flush();
   }));
 
+
+  it('sets expanded state when right arrow key is pressed', () => {
+    // given
+    setupWorkspace(component, fixture);
+    let element = component.workspace.getElement('subfolder');
+    component.uiState.selectedElement = element;
+    component.uiState.setExpanded(element.path, false);
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_RIGHT});
+
+    // then
+    let expandedState = component.uiState.isExpanded(element.path);
+    expect(expandedState).toBeTruthy();
+  });
+
+  it('keeps expanded state when right arrow key is pressed', () => {
+    // given
+    setupWorkspace(component, fixture);
+    let element = component.workspace.getElement('subfolder');
+    component.uiState.selectedElement = element;
+    component.uiState.setExpanded(element.path, true);
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_RIGHT});
+
+    // then
+    let expandedState = component.uiState.isExpanded(element.path);
+    expect(expandedState).toBeTruthy();
+  });
+
+  it('sets collapsed state when left arrow key is pressed', () => {
+    // given
+    setupWorkspace(component, fixture);
+    let element = component.workspace.getElement('subfolder');
+    component.uiState.selectedElement = element;
+    component.uiState.setExpanded(element.path, true);
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_LEFT});
+
+    // then
+    let expandedState = component.uiState.isExpanded(element.path);
+    expect(expandedState).toBeFalsy();
+  });
+
+  it('keeps collapsed state when left arrow key is pressed', () => {
+    // given
+    setupWorkspace(component, fixture);
+    let element = component.workspace.getElement('subfolder');
+    component.uiState.selectedElement = element;
+    component.uiState.setExpanded(element.path, false);
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_LEFT});
+
+    // then
+    let expandedState = component.uiState.isExpanded(element.path);
+    expect(expandedState).toBeFalsy();
+  });
+
+  it('selects the next sibling element when the down arrow key is pressed', async(() => {
+    // given
+    setupWorkspace(component, fixture);
+    component.uiState.selectedElement = nonExecutableFile;
+    fixture.detectChanges();
+
+    // when
+    sidenav.query(By.css('nav-tree-viewer')).triggerEventHandler('keyup', { key: KEY_DOWN});
+
+    // then
+    expect(component.uiState.selectedElement).toEqual(tclFile);
```

Comment:
test requires tclFile to be the next sibling of nonExecutableFile, this could change or rather is not really guaranteed by setupWorkspace. maybe it's possible to get the first element that has a sibling and us that or name the files such that it is clear that they are (and should always be) siblings.